### PR TITLE
Integrate and configure pallet_elections_phragmen in runtime

### PR DESCRIPTION
**Pull request by asiniscalchi**
### **PR Type**
enhancement, configuration changes

___

### **Description**
- Integrated `pallet_elections_phragmen` into the runtime, allowing for election functionalities.
- Added configuration for `pallet_elections_phragmen`, including parameters for candidacy bond, term duration, and voting lock period.
- Updated benchmarks to include `pallet_collective` and `pallet_elections_phragmen`.
- Modified `Cargo.toml` to include `pallet-elections-phragmen` in dependencies and feature sets.

___

### **Changes walkthrough** 📝
<table><thead><tr><th></th><th align="left">Relevant files</th></tr></thead><tbody><tr><td><strong>Enhancement</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>benchmarks.rs</strong><dd><code>Add benchmarks for collective and elections pallets</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/benchmarks.rs

<li>Added benchmarks for <code>pallet_collective</code> and <code>pallet_elections_phragmen</code>.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/723/files#diff-e1771479a1102a3e8954125c207e6f16ff6a00db135b75ea674751af1443fd1e">+2/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>lib.rs</strong><dd><code>Integrate elections pallet into runtime</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/lib.rs

- Integrated `pallet_elections_phragmen` into the runtime.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/723/files#diff-003c87018a12c01c5266d5e1f3aba274f64cc6651fdb6b34abb88a6d1a10d073">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Configuration changes</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>election_phragmen.rs</strong><dd><code>Add configuration for pallet_elections_phragmen</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/election_phragmen.rs

<li>Introduced configuration for <code>pallet_elections_phragmen</code>.<br> <li> Defined parameters such as <code>CandidacyBond</code>, <code>TermDuration</code>, and <br><code>VotingLockPeriod</code>.<br> <li> Implemented <code>pallet_elections_phragmen::Config</code> for <code>Runtime</code>.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/723/files#diff-4c3c0cd98ab5e5f679d0499216562bbbe6dc5638e76d05b9543343a7caff6da1">+55/-0</a>&nbsp; &nbsp; </td>

</tr>                    

<tr>
  <td>
    <details>
      <summary><strong>mod.rs</strong><dd><code>Include election_phragmen module in configurations</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/src/configs/mod.rs

- Added `election_phragmen` module to the configurations.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/723/files#diff-450b756d1af9090dd62cc432c3c21a1a4f655d52ee8b477c3d81e1038434d68b">+1/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr><tr><td><strong>Dependencies</strong></td><td><table>
<tr>
  <td>
    <details>
      <summary><strong>Cargo.toml</strong><dd><code>Add elections pallet to Cargo.toml dependencies</code>&nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; &nbsp; </dd></summary>
<hr>

runtime/laos/Cargo.toml

<li>Added <code>pallet-elections-phragmen</code> to dependencies.<br> <li> Included <code>pallet-elections-phragmen</code> in various feature sets.

</details>

  </td>
  <td><a href="https://github.com/freeverseio/laos/pull/723/files#diff-9b329717b96857f6884c9b9a32fc471267d6f9c0acc1a3b0e4758d73de6d34e5">+4/-0</a>&nbsp; &nbsp; &nbsp; </td>

</tr>                    
</table></td></tr></tr></tbody></table>

___

> 💡 **PR-Agent usage**:
>Comment `/help` on the PR to get a list of all available PR-Agent tools and their descriptions

### CODE REVIEW

File: runtime/laos/src/configs/election_phragmen.rs [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+#[cfg(feature = "fast-mode")]
+pub const ELECTION_VOTING_LOCK_DURATION: BlockNumber = 4 * HOURS;
+#[cfg(not(feature = "fast-mode"))]
+pub const ELECTION_VOTING_LOCK_DURATION: BlockNumber = 28 * DAYS;
+
+parameter_types! {
+	pub const CandidacyBond: Balance = 1000 * UNIT;
+	pub TermDuration: BlockNumber = TERM_DURATION;
+	pub VotingLockPeriod: BlockNumber = ELECTION_VOTING_LOCK_DURATION;
+	pub const DesiredMembers: u32 = 7;
```
Comment (magecnion) [inline runtime/laos/src/configs/election_phragmen.rs:26]: review this number

File: runtime/laos/src/configs/election_phragmen.rs [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	pub const DesiredRunnersUp: u32 = 20;
```
Comment (magecnion) [inline runtime/laos/src/configs/election_phragmen.rs:27]: review this number

File: runtime/laos/src/configs/election_phragmen.rs [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+};
+use frame_support::parameter_types;
+#[cfg(not(feature = "fast-mode"))]
+use parachains_common::DAYS;
+#[cfg(feature = "fast-mode")]
+use parachains_common::HOURS;
+use polkadot_runtime_common::CurrencyToVote;
+
+#[cfg(feature = "fast-mode")]
+pub const TERM_DURATION: BlockNumber = 4 * HOURS;
```
Comment (magecnion) [inline runtime/laos/src/configs/election_phragmen.rs:13]: in order to test locally we might want even lower values for `fast-mode`, what do you think?

File: runtime/laos/src/configs/election_phragmen.rs [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	/// Number of members to elect.
+	type DesiredMembers = DesiredMembers;
+	/// Number of runners_up to keep.
+	type DesiredRunnersUp = DesiredRunnersUp;
+	type InitializeMembers = Council;
+	type LoserCandidate = Treasury;
+	type MaxCandidates = MaxCandidates;
+	type MaxVoters = MaxVoters;
+	type MaxVotesPerVoter = MaxVotesPerVoter;
+	type PalletId = ();
```
Comment (magecnion) [inline runtime/laos/src/configs/election_phragmen.rs:50]: in the rococo runtime they add some id: https://github.com/paritytech/polkadot-sdk/blob/0bb6249268c0b77d2834640b84cb52fdd3d7e860/polkadot/runtime/rococo/src/lib.rs#L1588-L1589
we might want to do the same

**Review by luispdm — CHANGES_REQUESTED** [review]

File: runtime/laos/src/configs/election_phragmen.rs (hunk 3 above)
Comment (luispdm) [reply runtime/laos/src/configs/election_phragmen.rs:13]: +1. I suggest a few minutes, or even better, a few blocks. Same thing for `ELECTION_VOTING_LOCK_DURATION`

File: runtime/laos/src/configs/election_phragmen.rs (hunk 4 above)
Comment (luispdm) [reply runtime/laos/src/configs/election_phragmen.rs:50]: +1. We've done a similar thing for the treasury pallet -> https://github.com/freeverseio/laos/pull/716/files#diff-094b4a74f7e7a446c70dd2ca710d6a7c87a1c46369ec9a8125621479e247219aR30

File: runtime/laos/src/configs/election_phragmen.rs (hunk 3 above)
Comment (asiniscalchi) [reply runtime/laos/src/configs/election_phragmen.rs:13]: a bit complicated to propose, vote, reject, etc etc in a couple of blocks

**Comment by magecnion** [comment]
This https://github.com/freeverseio/laos/pull/751 should fix CI issues and add missing `PalletId` configuration param

**Pull request merged**